controllers/activemqartemis_controller_test.go

```diff
@@ -799,6 +799,37 @@ var _ = Describe("artemis controller", func() {
 	})
 
 	Context("Console config test", func() {
+		It("checking console target service port name for metrics", Label("console-expose-metrics"), func() {
+			By("Deploying a broker with console exposed")
+			brokerCr, createdBrokerCr := DeployCustomBroker(defaultNamespace, func(candidate *brokerv1beta1.ActiveMQArtemis) {
+				candidate.Spec.DeploymentPlan.Size = 2
+				candidate.Spec.DeploymentPlan.ReadinessProbe = &corev1.Probe{
+					InitialDelaySeconds: 1,
+					PeriodSeconds:       1,
+					TimeoutSeconds:      5,
+				}
+				candidate.Spec.Console.Expose = true
+			})
+
+			ssKey := types.NamespacedName{
+				Name:      namer.CrToSS(brokerCr.Name),
+				Namespace: defaultNamespace,
+			}
+			currentSS := &appsv1.StatefulSet{}
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, ssKey, currentSS)).Should(Succeed())
+				g.Expect(len(currentSS.Spec.Template.Spec.Containers[0].Ports)).To(Equal(2))
+				g.Expect(currentSS.Spec.Template.Spec.Containers[0].Ports[0].Name).To(Equal("wconsj-0"))
+				g.Expect(currentSS.Spec.Template.Spec.Containers[0].Ports[0].ContainerPort).To(Equal(int32(8161)))
+				g.Expect(currentSS.Spec.Template.Spec.Containers[0].Ports[0].Protocol).To(Equal(corev1.ProtocolTCP))
+				g.Expect(currentSS.Spec.Template.Spec.Containers[0].Ports[1].Name).To(Equal("wconsj-1"))
```

Comment:
I'm a bit confused, why do we need more than one port 8161 in the broker container spec?

Reply:
It depends on deploymentPlan: size. If size is 2, there will be two pods available for collecting metrics from. So we need to expose all the ports in the spec so that prometheus can collect each of them.

Reply:
For Hawtio Online we only added one port (`jolokia`) for all pods of a statefulset to support discovery, If multiple ports are defined in the broker container spec then each pods of the statefulset have multiple ports. I don't understand why the prometheus service monitor should require to define multiple ports. Is that a requirement from the prometheus doc?

Reply:
The service monitor collects metrics from a list of container ports. Not sure if *jolokia* port does the same thing but that's requested by some user. See https://issues.redhat.com/browse/ENTMQBR-5228

Reply:
I reviewed [ENTMQBR-5228](https://issues.redhat.com/browse/ENTMQBR-5228) and I'm not able to find where it is requested to add multiple ports for the console in each broker container.

See the example in the prometheus-operator getting-started guide:
https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/getting-started.md

One port defined in the statefulset for all 3 replicas
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: example-app
spec:
  replicas: 3
  selector:
    matchLabels:
      app: example-app
  template:
    metadata:
      labels:
        app: example-app
    spec:
      containers:
      - name: example-app
        image: fabxc/instrumented_app
        ports:
        - name: web
          containerPort: 8080
```
```
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name: example-app
  labels:
    team: frontend
spec:
  selector:
    matchLabels:
      app: example-app
  podMetricsEndpoints:
  - port: web
```

Reply:
I came across the same doc and you bet me to the comment. I bet the service monitor queries the stateful set metadata to find the existing set of service endpoints that host that port. Ie, it knows about a stateful set.

we man need a round trip test to validate fully, but it seems a single port is sufficient, the service exposes it as necessary based on the replicas.